src/Symfony/Component/DependencyInjection/Tests/LazyStringTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\DependencyInjection\LazyString;
+use Symfony\Component\ErrorHandler\ErrorHandler;
+
+class LazyStringTest extends TestCase
+{
+    public function testLazyString()
+    {
+        $count = 0;
+        $s = LazyString::fromCallable(function () use (&$count) {
+            return ++$count;
+        });
+
+        $this->assertSame(0, $count);
+        $this->assertSame('1', (string) $s);
+        $this->assertSame(1, $count);
```

Comment:
I'd add another assert to test the callable is called once and the counter does not increment anymore, wdyt?